test/aqua/operators/test_abelian_grouper.py

```diff
@@ -53,23 +53,19 @@ def test_abelian_grouper2(self):
                 self.assertTrue(op_1.commutes(op_2))
 
     def test_abelian_grouper3(self):
-        """ abelian grouper test 3 """
+        """Abelian grouper test 3"""
         paulis = (I ^ X) + (2 * X ^ X) + (3 * Z ^ Y)
         for fast in [True, False]:
             for use_nx in [True, False]:
                 grouped_sum = AbelianGrouper.group_subops(paulis, fast=fast, use_nx=use_nx)
                 self.assertEqual(len(grouped_sum), 2)
-                self.assertEqual(len(grouped_sum[0]), 2)
-                self.assertEqual(str(grouped_sum[0][0].primitive), 'IX')
-                self.assertEqual(grouped_sum[0][0].coeff, 1)
-                self.assertEqual(str(grouped_sum[0][1].primitive), 'XX')
-                self.assertEqual(grouped_sum[0][1].coeff, 2)
-                self.assertEqual(len(grouped_sum[1]), 1)
-                self.assertEqual(str(grouped_sum[1][0].primitive), 'ZY')
-                self.assertEqual(grouped_sum[1][0].coeff, 3)
+                self.assertListEqual([str(op.primitive) for op in grouped_sum[0]], ['IX', 'XX'])
+                self.assertListEqual([op.coeff for op in grouped_sum[0]], [1, 2])
+                self.assertListEqual([str(op.primitive) for op in grouped_sum[1]], ['ZY'])
+                self.assertListEqual([op.coeff for op in grouped_sum[1]], [3])
 
     def test_abelian_grouper4(self):
-        """ abelian grouper test 4 """
+        """Abelian grouper test 4"""
         paulis = X + (2 * Y) + (3 * Z)
```

Comment:
Does this test test different functionality or code parts than `test_abelian_grouper3`, or is it just a different Pauli string?

Reply:
It is just different pauli strings. I will make is a sub test.